### PR TITLE
[Add] add Rocm ep doc

### PR DESCRIPTION
**Pull request by PeixuanZuo**
**Description**: Describe your changes.
add rocm ep doc.
Perview Github page : https://ytaous.github.io/onnxruntime/

Added items
Home page
- [ ] Update matrix on home page: https://ytaous.github.io/onnxruntime/
- [ ] Link "Installation Instruction Follow build instructions from here" to onnxruntime build with ROCm(inference) page.(This link now point to preview page, will change later)

Installation
- [ ] Update installation page: https://ytaous.github.io/onnxruntime/docs/install/#training-install-table-for-all-languages

Build onnxruntime
- [ ] Add onnxruntime build with ROCm(inference):  https://ytaous.github.io/onnxruntime/docs/build/eps.html#amd-rocm
- [ ] Update onnxruntime build with ROCm(training): https://ytaous.github.io/onnxruntime/docs/build/training.html#gpu--rocm

ExecutionProvider
- [ ] Add ROCm ExecutionProvider navitagation: https://ytaous.github.io/onnxruntime/docs/execution-providers/
- [ ] Add ROCm-ExecutionProvider page: https://ytaous.github.io/onnxruntime/docs/execution-providers/ROCm-ExecutionProvider.html

Performance
- [ ] Add enable_rocm_profiling introcduction: https://ytaous.github.io/onnxruntime/docs/performance/tune-performance.html#profiling-and-performance-report
- [ ] Add graph-optimation for ROCm ep: https://ytaous.github.io/onnxruntime/docs/performance/graph-optimizations.html#extended-graph-optimizations

### CODE REVIEW

File: docs/execution-providers/ROCm-ExecutionProvider.md
Comment (cloudhan) [inline docs/execution-providers/ROCm-ExecutionProvider.md:12]: I think for "graph optimization", only MIGraphX EP will exhibit. More appropriately, this should be:

> The ROCm Execution Provider enables hardware accelerated computation on AMD ROCm-enabled GPUs.

analogs to docs for CUDA EP.

File: docs/execution-providers/index.md
Comment (cloudhan) [inline docs/execution-providers/index.md:37]: Shall we move this upward, so that it is layouted as

ROCm
MIGraphX

Comment (cloudhan) [reply docs/execution-providers/index.md:37]: AMD ROCm -> AMD ROCm (*preview*)

@faxu Can you confirm we are still in *preview* status?

Comment (ytaous) [reply docs/execution-providers/index.md:37]: Should be preview - good catch.
After it's stabilized, we'll remove preview later, thx.
We'll evaluate it when rocm5.3 is released.

File: css/custom.css
Comment (faxu) [inline css/custom.css:649]: Instead of adjusting the % for .col-lg-2dot5, you should probably just update the class on the main page matrix to col-lg-2dot4 for each of the EP selections

File: docs/build/training.md
Comment (faxu) [inline docs/build/training.md:85]: For consistency, can you fix capitalization of ROCM to ROCm?

File: docs/execution-providers/ROCm-ExecutionProvider.md
Comment (faxu) [inline docs/execution-providers/ROCm-ExecutionProvider.md:5]: The nav order for EP pages should be alphabetical, so this should follow OpenVINO. Please update nav_order of all other EP pages as well to shift them correctly.

File: docs/execution-providers/ROCm-ExecutionProvider.md
Comment (faxu) [inline docs/execution-providers/ROCm-ExecutionProvider.md:29]: The default branch is now called "main"
If you saw this from another EP page, appreciate if you can update it there as well

File: docs/execution-providers/ROCm-ExecutionProvider.md
Comment (faxu) [inline docs/execution-providers/ROCm-ExecutionProvider.md:54]: Is there any ROCm specific performance tuning on that page? If not, I suggest to not list this link.

File: docs/execution-providers/index.md
Comment (faxu) [reply docs/execution-providers/index.md:37]: Please list this under the AMD MIGraphX entry and add (Preview)

File: docs/execution-providers/ROCm-ExecutionProvider.md
Comment (ytaous) [reply docs/execution-providers/ROCm-ExecutionProvider.md:54]: Yes, there's new profiler included - https://github.com/microsoft/onnxruntime/pull/10911
User just need to include "--enable_rocm_profiling" in build option, useful for perf tuning. 
FYI

File: docs/execution-providers/ROCm-ExecutionProvider.md
Comment (cloudhan) [inline docs/execution-providers/ROCm-ExecutionProvider.md:45]: It is weird that session options is abbreviated as `sf`, can we rename it to `so` or better
```cpp
Ort::ThrowOnError(OrtSessionOptionsAppendExecutionProvider_ROCm(Ort::SessionOptions{}, device_id));
```
since we are not passing any special optoin here.

**Review by ytaous — APPROVED** [review]
LGTM - perhaps address the last comment and wait for approval from others before merge.
Thanks.

**Review by faxu — APPROVED** [review]

**Comment by faxu** [comment]
Thanks for the fixes. Let me know when it's ready to merge.

**Review by cloudhan — APPROVED** [review]

**Comment by ytaous** [comment]
> Thanks for the fixes. Let me know when it's ready to merge.

Hi, @faxu - can u pls merge it now, thx a lot

**Pull request merged**
